### PR TITLE
Feature/emoji

### PR DESCRIPTION
**Pull request by wjian23**

### CODE REVIEW

File: widgets/src/text.rs [hunk 1]
```diff
@@ -66,38 +66,73 @@ impl Render for Text {
...
-      PathPaintStyle::Stroke(stroke) => {
-        painter
-          .set_strokes(stroke.clone())
-          .stroke_text(text, bounds, self.overflow);
-      }
+    let font_size = self.text_style.font_size.into_pixel().value();
+    let rc = visual_glyphs.visual_rect();
+    let font_db = ctx.typography_store().font_db().clone();
+    let painter = ctx.painter();
+    visual_glyphs.visual_rect();
```
Comment (M-Adoo) [inline widgets/src/text.rs:81]: unused statement?

File: text/src/text_render.rs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    if let Some(face) = face {
+      let unit = face.units_per_em() as f32;
+      let scale = font_size / unit;
+      if let Some(path) = face.outline_glyph(g.glyph_id) {
+        let ts = Transform::scale(1., -1.)
+          .then_translate((0., unit).into())
+          .then_scale(scale, scale)
+          .then_translate(Vector::new(g.bound.min_x(), g.bound.min_y()));
+        let path = Path::from(path).transform(&ts);
```
Comment (M-Adoo) [inline text/src/text_render.rs:72]: Set transform to painter, not transform the path, we'll cache the glyph path in the future and keep a more precision path for the painter backend.

File: core/src/builtin_widgets/fitted_box.rs [hunk 3]
```diff
@@ -55,19 +55,19 @@ impl Render for FittedBox {
...
-      let path = Path::rect(&Rect::from(rect.size));
-      ctx.painter().clip(path);
+    let Transform { m11: x, m22: y, .. } = self.scale_cache.get();
+    if matches!(self.box_fit, BoxFit::Cover) {
+      let size = ctx.box_size().unwrap();
+      let child_size = ctx
+        .single_child_box()
+        .expect("Should always have a single child")
+        .size;
+      if size.width < child_size.width * x || size.height < child_size.height {
```
Comment (M-Adoo) [inline core/src/builtin_widgets/fitted_box.rs:65]: height not scale

File: painter/src/painter.rs [hunk 4]
```diff
@@ -452,14 +452,42 @@ impl Painter {
...
 
-  pub fn draw_img(&mut self, img: ShareResource<PixelImage>) -> &mut Self {
-    self
-      .rect(&Rect::from_size(Size::new(
-        img.width() as f32,
-        img.height() as f32,
-      )))
-      .set_brush(img)
-      .fill();
+  /// Draw the image
```
Comment (M-Adoo) [inline painter/src/painter.rs:455]: Does this API do drawing `src_rect` from image to `dst_rect` in the painter context?

**Review by M-Adoo — APPROVED** [review]

**Review by sologeek — APPROVED** [review]

**Pull request merged**